src/components/DatasetQueryView.jsx

```diff
@@ -0,0 +1,190 @@
+import LastPage from '@material-ui/icons/LastPage';
+import Remove from '@material-ui/icons/Remove';
+import SaveAlt from '@material-ui/icons/SaveAlt';
+import Search from '@material-ui/icons/Search';
+import ViewColumn from '@material-ui/icons/ViewColumn';
+
+import { runQuery } from 'actions/index';
+
+const tableIcons = {
+  Add: React.forwardRef((props, ref) => <AddBox {...props} ref={ref} />),
+  Check: React.forwardRef((props, ref) => <Check {...props} ref={ref} />),
+  Clear: React.forwardRef((props, ref) => <Clear {...props} ref={ref} />),
+  Delete: React.forwardRef((props, ref) => <DeleteOutline {...props} ref={ref} />),
+  DetailPanel: React.forwardRef((props, ref) => <ChevronRight {...props} ref={ref} />),
+  Edit: React.forwardRef((props, ref) => <Edit {...props} ref={ref} />),
+  Export: React.forwardRef((props, ref) => <SaveAlt {...props} ref={ref} />),
+  Filter: React.forwardRef((props, ref) => <FilterList {...props} ref={ref} />),
+  FirstPage: React.forwardRef((props, ref) => <FirstPage {...props} ref={ref} />),
+  LastPage: React.forwardRef((props, ref) => <LastPage {...props} ref={ref} />),
+  NextPage: React.forwardRef((props, ref) => <ChevronRight {...props} ref={ref} />),
+  PreviousPage: React.forwardRef((props, ref) => <ChevronLeft {...props} ref={ref} />),
+  ResetSearch: React.forwardRef((props, ref) => <Clear {...props} ref={ref} />),
+  Search: React.forwardRef((props, ref) => <Search {...props} ref={ref} />),
+  SortArrow: React.forwardRef((props, ref) => <ArrowUpward {...props} ref={ref} />),
+  ThirdStateCheck: React.forwardRef((props, ref) => <Remove {...props} ref={ref} />),
+  ViewColumn: React.forwardRef((props, ref) => <ViewColumn {...props} ref={ref} />),
+};
+
+const styles = theme => ({
+  root: {
+    border: `1px solid ${theme.palette.primary.dark}`,
+    borderRadius: `${theme.shape.borderRadius}px ${theme.shape.borderRadius}px 0 0`,
+    boxShadow: 'none',
+    marginTop: theme.spacing(3),
+    maxWidth: 1400,
+    overflowX: 'auto',
+    width: '100%',
+    overflowWrap: 'break-word',
+  },
+  table: {
+    borderRadius: `${theme.shape.borderRadius}px ${theme.shape.borderRadius}px 0 0`,
+    minWidth: 700,
+  },
+  row: {
+    borderRadius: `${theme.shape.borderRadius}px ${theme.shape.borderRadius}px 0 0`,
+  },
+  searchIcon: {
+    color: theme.palette.primary.main,
+    width: 22,
+    height: '100%',
+    position: 'absolute',
+    pointerEvents: 'none',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    paddingLeft: theme.spacing(3),
+  },
+  searchInput: {
+    paddingTop: theme.spacing(1.5),
+    paddingLeft: theme.spacing(6),
+  },
+});
+
+export class DatasetQueryView extends React.PureComponent {
+  static propTypes = {
+    dispatch: PropTypes.func.isRequired,
+    queryResults: PropTypes.object,
+    token: PropTypes.string,
+  };
+
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch(
+      runQuery(
+        'broad-jade-my',
```

Comment:
nit: in the future add a TODO so clarifying upcoming work to replace hardcoded values